lib/ui/widget/dimmed_image_background.dart

```diff
@@ -0,0 +1,76 @@
+import 'package:flutter/material.dart';
+
+import 'blurred_background.dart';
+import 'dimmed_background.dart';
+
+class DimmedImageBackground extends StatefulWidget {
```

Comment:
It's an image background why we have to create it as `StatefulWidget` why not create it `StatelessWidget` and handle the state (shouldAnimate, animationDuration) on parent screen for example in `SplashScreen` 🤔

Reply:
> 

@Wadeewee please check Mark's [comment](https://github.com/nimblehq/ic-flutter-avishek/pull/32#discussion_r1257686665) regarding keeping the animation states inside the widget.

Regarding separating the widgets into different files, it's for separation of concern, and also, they will be reused by other screens.

Reply:
> Regarding your POW here 👇 since `0.04 - 0.07` sec. It should be the animation when navigating to sign-in screen 🤔
> 
>  Screen.Recording.2023-07-06.at.6.24.43.PM.mov

@Wadeewee this is only for the splash screen. The login form will appear on the same screen during the animation of the logo.

Reply:
>@Wadeewee please check Mark's https://github.com/nimblehq/ic-flutter-avishek/pull/32#discussion_r1257686665 >regarding keeping the animation states inside the widget.

>Regarding separating the widgets into different files, it's for separation of concern, and also, they will be reused by >other screens.

@AVI5HEK May I know which screen we can reuse them. I saw you target `LoginScreen` instead of `SplashScreen` on this  PR 🤔 https://github.com/nimblehq/ic-flutter-avishek/pull/33/files#diff-e61eb31d013d12616f5532636a88cfa63631dda8f7829e5424e68542214d1608. Are you using the same screen for `splash` and `login` screen

Reply:
@Wadeewee it'll be used in home, reset password, and thank you screens.

Reply:
Got it 🚀